packages/widget/src/controllers/wallet-manager/token-balance.ts

```diff
@@ -100,4 +119,25 @@ export class TokenBalanceController implements ReactiveController {
       this.host.requestUpdate();
     }.bind(this)();
   };
+
+  suscribeSubstrateBalanceUpdate = (resource: SubstrateResource): void => {
+    const { substrateProvider: apiPromise, accounts } = this.walletContext.value
+      ?.substrateWallet as SubstrateWallet;
+
+    const accountAddress = accounts![0].address;
+
+    void async function (this: TokenBalanceController) {
+      const balance = await getAssetBalance(
+        apiPromise as ApiPromise,
+        (resource as unknown as { assetID: number }).assetID,
+        accountAddress
+      );
+      this.loadingBalance = true;
+      this.host.requestUpdate();
+      this.loadingBalance = false;
+      this.balance = BigNumber.from(BigInt(balance.balance.toString()));
```

Comment:
what is the type of balance.balance?

Reply:
u128

Reply:
change name to tokenBalance to avoid having `balance.balance`.